### PR TITLE
CEM-2409- create component ItemsRedemptionPoints

### PR DESCRIPTION
**Pull request by roggc**
redemptionPoints2

### CODE REVIEW

File: src/Containers/ItemsRedemptionPoints/ItemsRedemptionPoints.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    const closeModal=()=>{
+        setIsModalShown(false)
+    }
+
+    /**
+     * do business logic with the state of dataItems set by the user
+     * @param dataItems {IData[]} - the state of dataItems as modified by the user interaction
+     */
+    const applyChanges=(dataItems:IData[])=>{
+        // do something with dataItems
```
Comment (ralph-dev) [inline src/Containers/ItemsRedemptionPoints/ItemsRedemptionPoints.stories.tsx:56]: Shouldn't we see the value of the switch update to the value inputted?

File: src/Containers/ItemsRedemptionPoints/ItemsRedemptionPoints.tsx [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+import {Paragraph as P} from '@Text'
+import {MainTheme} from '@Themes'
+import { Main, MainInterface } from '@Utils/BaseStyles';
+import {Mixins} from '@Utils'
+import {Button} from '@Inputs/Button/Button'
+import {Input} from '@Inputs/Input/Input'
+import {Card} from '@Containers/Card/Card'
+
+const DECIMAL_BASE=10
+const TIMES_VERTICAL_PADDING=1.5
```
Comment (ralph-dev) [inline src/Containers/ItemsRedemptionPoints/ItemsRedemptionPoints.tsx:13]: Why do you need to times though, what does the vertical padding do?

File: src/Containers/ItemsRedemptionPoints/ItemsRedemptionPoints.tsx [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+
+const ItemRedemptionPoints:React.FC<IItemRedemptionPoints>=({index,dataItems,setDataItems}):React.ReactElement=>{
+
+    /**
+     * sets the redemptionPoints value based on the value of the input field
+     * @param event {React.ChangeEvent<HTMLInputElement>} - the event fired by input element
+     */
+    const updateRedemptionPoints=({target}:React.ChangeEvent<HTMLInputElement>)=>{
+        let value=parseInt(target.value,DECIMAL_BASE)
+        if(value<0) value=0
```
Comment (ralph-dev) [inline src/Containers/ItemsRedemptionPoints/ItemsRedemptionPoints.tsx:147]: 0 means NOT REDEEMABLE

**Review by ralph-dev — COMMENTED** [review]
Looking good!

**Comment by ralph-dev** [comment]
Are you using the green dot status component from react UI?

File: src/Containers/ItemsRedemptionPoints/ItemsRedemptionPoints.stories.tsx (hunk 1 above)
Comment (roggc) [reply src/Containers/ItemsRedemptionPoints/ItemsRedemptionPoints.stories.tsx:56]: can you rephrase this please?

Comment (ralph-dev) [reply src/Containers/ItemsRedemptionPoints/ItemsRedemptionPoints.stories.tsx:56]: shouldn't we pass a batchUpdate Function

File: src/Containers/ItemsRedemptionPoints/ItemsRedemptionPoints.tsx [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+    applyToAllItemsText:string;
+}
+
+export const ItemsRedemptionPoints:React.FC<IItemsRedemptionPointsProps>=({modalProps,titleText,titleDescription,cancelButtonText,applyButtonText,batchUpdateButtonText,onClickApplyButton,onClickCancelButton,data,...props}):React.ReactElement=>{
+    const [isShownBatchUpdateModal,setIsShownBatchUpdateModal]=useState(false)
+
+    /**
+     * this is used to store info about the ItemRedemptionPoints child components
+     */
+    const infoRef=useRef<IReferenceValues[]>(new Array(data.length))
```
Comment (ralph-dev) [inline src/Containers/ItemsRedemptionPoints/ItemsRedemptionPoints.tsx:55]: Why not an array of values in state?

**Review by ralph-dev — APPROVED** [review]
Lgtm

File: src/Containers/ItemsRedemptionPoints/ItemsRedemptionPoints.tsx (hunk 4 above)
Comment (roggc) [reply src/Containers/ItemsRedemptionPoints/ItemsRedemptionPoints.tsx:55]: I don't know, I am kind used to do it this way. I don't like too much to uplift state.

Comment (ralph-dev) [reply src/Containers/ItemsRedemptionPoints/ItemsRedemptionPoints.tsx:55]: You know when u use REF UI won't update, but when you update State UI will update right?

Comment (ralph-dev) [reply src/Containers/ItemsRedemptionPoints/ItemsRedemptionPoints.tsx:55]: Ref is not for data, it's for referencing components please fix this

File: src/Containers/ItemsRedemptionPoints/ItemsRedemptionPoints.tsx [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+     * closes batch update modal
+     */
+    const closeBatchUpdateModal=()=>{
+        setIsBatchUpdateModalShown(false)
+    }
+
+    /**
+     * applyes the value redemptionPoints to all data items and closes batch update modal
+     */
+    const doBatchUpdate=()=>{
```
Comment (ralph-dev) [inline src/Containers/ItemsRedemptionPoints/ItemsRedemptionPoints.tsx:239]: onBatchUpdate

**Review by ralph-dev — APPROVED** [review]
LGTM
